internal/npm_install/generate_build_file.ts

```diff
@@ -854,99 +849,116 @@ function findFile(pkg: Dep, m: string) {
-    named_module_srcs = [
-        ${namedSources.map((f: string) => `"//:node_modules/${pkg._dir}/${f}",`).join('\n        ')}
-    ],`;
-  }
-
-  let filesStarlark = '';
-  if (files.length) {
-    filesStarlark = `
-    # ${pkg._dir} package files (and files in nested node_modules)
-    srcs = [
+  function starlarkFiles(attr: string, files: string[], comment: string = '') {
+    return `
+    ${comment ? comment + '\n    ' : ''}${attr} = [
         ${files.map((f: string) => `"//:node_modules/${pkg._dir}/${f}",`).join('\n        ')}
     ],`;
   }
 
-  let nestedNodeModulesStarlark = '';
-  if (nestedNodeModules.length) {
-    nestedNodeModulesStarlark = `
-    # ${pkg._dir} package files (and files in nested node_modules)
-    srcs = [
-        ${
-        nestedNodeModules.map((f: string) => `"//:node_modules/${pkg._dir}/${f}",`)
-            .join('\n        ')}
-    ],`;
-  }
+  const includedRunfiles = filterFiles(pkg._runfiles, INCLUDED_FILES);
 
-  let depsStarlark = '';
-  if (deps.length) {
-    const list = deps.map(dep => `"//${dep._dir}:${dep._name}__contents",`).join('\n        ');
-    depsStarlark = `
-    # flattened list of direct and transitive dependencies hoisted to root by the package manager
-    deps = [
-        ${list}
-    ],`;
-  }
+  // Files that are part of the npm package not including its nested node_modules
+  // (filtered by the 'included_files' attribute)
+  const pkgFiles = includedRunfiles.filter((f: string) => !f.startsWith('node_modules/'));
+  const pkgFilesStarlark = pkgFiles.length ? starlarkFiles('srcs', pkgFiles) : '';
 
-  let dtsStarlark = '';
-  if (dtsSources.length) {
-    dtsStarlark = `
-    # ${pkg._dir} package declaration files (and declaration files in nested node_modules)
-    srcs = [
-        ${dtsSources.map(f => `"//:node_modules/${pkg._dir}/${f}",`).join('\n        ')}
-    ],`;
-  }
+  // Files that are in the npm package's nested node_modules
+  // (filtered by the 'included_files' attribute)
+  const nestedNodeModules = includedRunfiles.filter((f: string) => f.startsWith('node_modules/'));
+  const nestedNodeModulesStarlark =
+      nestedNodeModules.length ? starlarkFiles('srcs', nestedNodeModules) : '';
+
+  // Files that have been excluded from the ${pkg._name}__files target above because
+  // they are filtered out by 'included_files' or because they are not valid runfiles
+  // See https://github.com/bazelbuild/bazel/issues/4327.
+  const notPkgFiles = pkg._files.filter(
+      (f: string) => !f.startsWith('node_modules/') && !includedRunfiles.includes(f));
+  const notPkgFilesStarlark = notPkgFiles.length ? starlarkFiles('srcs', notPkgFiles) : '';
+
+  // If the package is in the Angular package format returns list
+  // of package files that end with `.umd.js`, `.ngfactory.js` and `.ngsummary.js`.
+  // TODO(gmagolan): add UMD & AMD scripts to scripts even if not an APF package _but_ only if they
+  // are named?
+  const namedSources = isNgApfPackage(pkg) ?
```

Comment:
so many changes in this file - I'm assuming it's just a refactoring moving lines around, and not this many lines edited?

Reply:
Yup. Some refactoring here.